### PR TITLE
Add docs for @inline

### PR DESCRIPTION
**Pull request by robrichard**

### CODE REVIEW

**Comment by babangsund** [comment]
I like how you snuck the `pinned` in there. Bold! 😉

**Comment by babangsund** [comment]
> I like how you snuck the `pinned` in there. Bold! 😉

EDIT: 

To which I'd add that someone like @entria probably deserves being higher on the pinned list, considering their contributions and evangelism in the Relay community. 👍 

I could be ignorant here - there may be other deciding factors for being "pinned" worthy that I'm unaware of. 🙂 (In which case, please disregard my comment 😅)

File: docs/Modern-GraphQLInRelay.md [hunk 1]
```diff
@@ -89,11 +89,68 @@ fragment TodoApp_app on App {
 }
 ```
 
+### `@inline`
+
+By default Relay will only expose the data for fields explicitly requested by a [component's fragment](./fragment-container.html#createfragmentcontainer), which is known as [data masking](./thinking-in-relay#data-masking). Fragment data is unmasked for use in React components by `createFragmentContainer`. However, you may want to use fragment data in non-React functions that are called from React.
+
+Non-React functions can also take advantage of data masking. A fragment can be defined with the `@inline` directive and stored in a local variable. The non-React function can than "unmask" the data using the `readInlineData` function.
```
Comment (babangsund) [inline docs/Modern-GraphQLInRelay.md]: can **then** "unmask"

File: docs/Modern-GraphQLInRelay.md [hunk 2]
```diff
@@ -89,11 +89,68 @@ fragment TodoApp_app on App {
...
+```javascript
+// React Component
+function MyComponent({item}) {
+  function handleClick() {
+    processItemData(item);
+  }
+
+  return (
+    <button onClick={handleClick}>Process {item.title}</button>
+  )
```
Comment (babangsund) [inline docs/Modern-GraphQLInRelay.md]: Missing semi 😂

File: website/siteConfig.js [hunk 3]
```diff
@@ -54,7 +54,7 @@ const users = [
     caption: '1stdibs',
     image: '/img/logos/1stdibs.png',
     infoLink: 'https://www.1stdibs.com/',
-    pinned: false,
+    pinned: true,
```
Comment (kassens) [inline website/siteConfig.js]: Sneaky!

File: docs/Modern-GraphQLInRelay.md [hunk 4]
```diff
@@ -89,11 +89,68 @@ fragment TodoApp_app on App {
...
+  sendToThirdPartyApi({
+    title: item.title,
+    price: item.price,
+    creatorName: item.creator.name
+  });
+}
+```
+
+```javascript
+// React Component
```
Comment (jstejada) [inline docs/Modern-GraphQLInRelay.md]: we should probably show this component's fragment too

Comment (robrichard) [reply docs/Modern-GraphQLInRelay.md]: This component's fragment is a few lines down inside `createFragmentContainer`, did you have a different way of displaying it in mind?

**Comment by babangsund** [comment]
🎉

**Pull request closed without merging**